easydock/protonation.py

```diff
@@ -27,3 +33,46 @@ def read_protonate_chemaxon(fname):
             smi = mol.GetPropsAsDict().get('MAJORMS', None)
             if smi is not None:
                 yield smi, mol_name
+
+def chunk_into_n(smi_l: list[str], n: int):
+    smi_size = ceil(len(smi_l) / n)
+    return list(map(lambda x: smi_l[x * smi_size:x * smi_size + smi_size], list(range(n))))
+
+def dummy_protonate_dimorphite(input_output_fname: tuple[str, str]):
+    input_fname, output_fname = input_output_fname
+    dimorphite_run(smiles_file=input_fname, output_file=output_fname, max_variants=1, silent=True, min_ph=7.4, max_ph=7.4)
+
+def protonate_dimorphite(input_fname: str, output_fname: str, ncpu: int = 1):
+
+    with open(input_fname,'r') as input_file:
+        smi_l = input_file.readlines()
+
+    chunk_smi_l = chunk_into_n(smi_l, n=ncpu)
+
+    temp_fname_list = []
+    for chunk in chunk_smi_l:
+        with tempfile.NamedTemporaryFile(suffix='.smi',mode='w',encoding='utf-8', delete=False) as input_tmp:
+
+            input_tmp.write(''.join(chunk))
+            output_tmp = tempfile.NamedTemporaryFile(suffix='.smi',mode='w',encoding='utf-8', delete=False)
+            temp_fname_list.append((input_tmp.name, output_tmp.name))
+    
+    pool = Pool(ncpu)
+    pool.map(dummy_protonate_dimorphite, temp_fname_list)
+    pool.close()
+    pool.join()
+
+    with open(output_fname, 'wt') as output_file:
+        for temp_fname in temp_fname_list:
+            input_temp_fname, output_temp_fname = temp_fname
+
+            with open(output_temp_fname,'r') as output_smi:
+                output_file.write(''.join(output_smi))
```

Comment:
I'm not sure that this will work. `output_smi` will be a file descriptor. This should work:
```python
output_file.write(''.join(output_smi.read()))
```
It will work for not very large files, but we do not expect currently to process trillions of molecules